lib/crypto/base36.js

```diff
@@ -0,0 +1,30 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var crypto = require('crypto')
+
+function base36(len) {
+  var out = ''
+  while (out.length < len) {
+    var rand = crypto.randomBytes(len)
+    var randLen = rand.length
+    for (var i = 0; i < randLen; i++) {
+      var b = rand[i]
+      // 252-256 skews the base36 distribution, so skip those bytes
+      if (b < 252) {
```

Comment:
Yes, the bytes are 8 bits. Yes, if any of the bytes are out of the range, the output would be too short. There is an outer `while` loop to catch for that, and if so, to get some more random bytes and try again.

I settled on using the exact length for `randomBytes` since it means only 1 call in the majority of cases.